components/ai_chat/core/browser/conversation_handler.cc

```diff
@@ -884,33 +887,49 @@ void ConversationHandler::PerformAssistantGeneration(
       base::BindOnce(&ConversationHandler::OnEngineCompletionComplete,
                      weak_ptr_factory_.GetWeakPtr());
 
-  bool should_refine_page_content =
-      features::IsPageContentRefineEnabled() &&
-      page_content.length() > GetMaxContentLengthForModel(GetCurrentModel()) &&
-      input != l10n_util::GetStringUTF8(IDS_AI_CHAT_QUESTION_SUMMARIZE_PAGE);
-  if (should_refine_page_content && associated_content_delegate_) {
-    DVLOG(2) << "Asking to refine content, which is of length: "
-             << page_content.length();
+  // Determine if page content refinement is necessary
+  const bool is_refinement_enabled = features::IsPageContentRefineEnabled();
+  const uint32_t max_content_length =
+      GetMaxPageContentLengthFromModel(GetCurrentModel());
+  const bool is_content_too_long = page_content.length() > max_content_length;
+  const std::string summarize_page_question =
+      l10n_util::GetStringUTF8(IDS_AI_CHAT_QUESTION_SUMMARIZE_PAGE);
+  const bool is_not_summarize_request = input != summarize_page_question;
+
+  const bool should_refine_page_content =
+      is_refinement_enabled && is_content_too_long &&
+      is_not_summarize_request && associated_content_delegate_;
+
+  if (should_refine_page_content) {
```

Comment:
I also feel this block would feel easier to read if the expressions were in the conditional, rather than assigned to `should_refine_page_content`, and then checked in the conditional.